### DISCUSSION TITLE
Questions about dreambooth's logic in so-called new method

### DISCUSSION THREAD

**Discussion by J3m3** [TheLastBen/fast-stable-diffusion#862] [Q&A] [answered]
Hello wonderful community! I have some questions about the implementation of DreamBooth in this code repository.

1. What's the difference between the so-called "old method" and the so-called "new method"? I've tried to check it out, but I can't access the .ipynb file of the old method.

2. Checked [your reply on #616](https://github.com/TheLastBen/fast-stable-diffusion/discussions/616#discussioncomment-4236153), and wondering how that automatic class detection is possible. Maybe by "--image_captions_filename" option in the script in "fast-DreamBooth.ipynb"?
```
!accelerate launch /content/diffusers/examples/dreambooth/train_dreambooth.py \  
  $Textenc \  
  --image_captions_filename \  
  --save_starting_step=$stpsv \  
  --stop_text_encoder_training=$stptxt \  
  --save_n_steps=$stp \  
  --Session_dir=$SESSION_DIR \  
  --pretrained_model_name_or_path="$MODELT_NAME" \  
  --instance_data_dir="$INSTANCE_DIR" \  
  --output_dir="$OUTPUT_DIR" \  
  --instance_prompt="$PT" \  
  --seed=$Seed \  
  --resolution=$Res \  
  --mixed_precision=$precision \  
  --train_batch_size=1 \  
  --gradient_accumulation_steps=1 $GC \  
  --use_8bit_adam \  
  --learning_rate=2e-6 \  
  --lr_scheduler="polynomial" \  
  --lr_warmup_steps=0 \  
  --max_train_steps=$Training_Steps
```
3. So if the class is automatically detected, how should I enter the prompt? For example, let's say I've trained a specific style of painting with file names like dlsgk (1).jpg, dlsgk (2).jpg, ..., and so on. If I want to generate a wolf style of dlsgk, what prompt do I have to enter? (If any class name is not needed in the prompt, what is the reason?)

4. Checked [your reply on #543](https://github.com/TheLastBen/fast-stable-diffusion/discussions/543#discussioncomment-4217628), wondering if you found it out with experiments.

Thanks for your hard work and contribution! (and sorry for so many questions...)

**Comment by TheLastBen** [comment, marked as answer]
Style instance images : dlsgk (1).jpg, dlsgk (2).jpg ....
Training steps : 8000
Text encoder : 15%

After training, use the prompt : wolf dglsk
You can enhance the prompt for better result but that's the basic one.

The old method you need an instance prompt and class images and an instance prompt , and retraining the model breaks it.
